homeassistant/components/alarmdecoder/alarm_control_panel.py

```diff
@@ -108,75 +102,30 @@ async def async_added_to_hass(self):
-            self._state = STATE_ALARM_ARMED_NIGHT
+            self._attr_state = STATE_ALARM_ARMED_NIGHT
         elif message.armed_home:
-            self._state = STATE_ALARM_ARMED_HOME
+            self._attr_state = STATE_ALARM_ARMED_HOME
         else:
-            self._state = STATE_ALARM_DISARMED
-
-        self._ac_power = message.ac_power
-        self._alarm_event_occurred = message.alarm_event_occurred
-        self._backlight_on = message.backlight_on
-        self._battery_low = message.battery_low
-        self._check_zone = message.check_zone
-        self._chime = message.chime_on
-        self._entry_delay_off = message.entry_delay_off
-        self._programming_mode = message.programming_mode
-        self._ready = message.ready
-        self._zone_bypassed = message.zone_bypassed
-
-        self.schedule_update_ha_state()
-
-    @property
-    def name(self):
-        """Return the name of the device."""
-        return self._name
-
-    @property
-    def should_poll(self):
-        """Return the polling state."""
-        return False
-
-    @property
-    def code_format(self):
-        """Return one or more digits/characters."""
-        return FORMAT_NUMBER
-
-    @property
-    def state(self):
-        """Return the state of the device."""
-        return self._state
-
-    @property
-    def supported_features(self) -> int:
-        """Return the list of supported features."""
-        return SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_AWAY | SUPPORT_ALARM_ARM_NIGHT
-
-    @property
-    def code_arm_required(self):
-        """Whether the code is required for arm actions."""
-        return self._code_arm_required
-
-    @property
-    def extra_state_attributes(self):
-        """Return the state attributes."""
-        return {
-            "ac_power": self._ac_power,
-            "alarm_event_occurred": self._alarm_event_occurred,
-            "backlight_on": self._backlight_on,
-            "battery_low": self._battery_low,
-            "check_zone": self._check_zone,
-            "chime": self._chime,
-            "entry_delay_off": self._entry_delay_off,
-            "programming_mode": self._programming_mode,
-            "ready": self._ready,
-            "zone_bypassed": self._zone_bypassed,
-            "code_arm_required": self._code_arm_required,
+            self._attr_state = STATE_ALARM_DISARMED
+
+        self._attr_extra_state_attributes = {
+            "ac_power": message.ac_power,
+            "alarm_event_occurred": message.alarm_event_occurred,
+            "backlight_on": message.backlight_on,
+            "battery_low": message.battery_low,
+            "check_zone": message.check_zone,
+            "chime": message.chime_on,
+            "entry_delay_off": message.entry_delay_off,
+            "programming_mode": message.programming_mode,
+            "ready": message.ready,
+            "zone_bypassed": message.zone_bypassed,
+            "code_arm_required": self._attr_code_arm_required,
```

Comment:
This is already in the entity attributes, we shouldn't override it in the extra attributes